app/components/filmstrip.jsx

```diff
@@ -0,0 +1,120 @@
+import React from 'react';
+import { DISCIPLINES } from './disciplines';
+
+export default class Filmstrip extends React.Component {
+  constructor(props) {
+    super(props);
+
+    this.scrollLeft = this.scrollLeft.bind(this);
+    this.scrollRight = this.scrollRight.bind(this);
+  }
+
+  componentDidMount() {
+    this.filmstrip.style.height = `${this.strip.clientHeight}px`;
+    this.viewport.style.paddingBottom = `${this.viewport.offsetHeight - this.viewport.clientHeight}px`;
+  }
+
+  scrollLeft() {
+    this.adjustPos(-this.props.increment);
+  }
+
+  scrollRight() {
+    this.adjustPos(this.props.increment);
+  }
+
+  mangleFilterName(filterName) {
+    return filterName.replace(/\s+/g, '-');
+  }
+
+  selectFilter(filterName) {
+    this.props.onChange(filterName);
+  }
+
+  calculateClasses(filterName) {
+    const mangledFilterName = this.mangleFilterName(filterName);
+
+    const list = ['filmstrip--disciplines__discipline-card'];
+    list.push(`filmstrip--disciplines__discipline-card--discipline-${mangledFilterName}`);
+
+    if (this.props.value === filterName) {
+      list.push('filmstrip--disciplines__discipline-card--active');
+    }
+    if (!this.props.value && filterName === 'all') {
+      list.push('filmstrip--disciplines__discipline-card--active');
+    }
+    if (this.props.value === '' && filterName === 'all') {
+      list.push('filmstrip--disciplines__discipline-card--active');
+    }
+
+    return list.join(' ');
+  }
+
+  adjustPos(increment) {
```

Comment:
Perhaps a bit picky, but would you mind using names that are a bit more descriptive, e.g. adjustPosition. Same for all the other "pos".